tools/build-tools/src/typeValidator/packageJson.ts

```diff
@@ -41,14 +44,39 @@ function createSortedObject<T>(obj:Record<string,T>): Record<string,T>{
     return sortedDeps;
 }
 
-export function getPackageDetails(packageDir: string): PackageDetails {
+function safeParse(json: string, error: string){
+    try{
+        return JSON.parse(json);
+    }catch{
+        throw new Error(error);
+    }
+}
+
+export async function getPackageDetailsOrThrow(packageDir: string, updateOptions?: {cwd?: string}):  Promise<PackageDetails> {
+    const result = await getPackageDetails(packageDir, updateOptions);
+    if(result.error !== undefined){
+        throw new Error(result.error);
+    }
+    return result;
+}
+
+export async function getPackageDetails(packageDir: string, updateOptions?: {cwd?: string}): Promise<PackageDetails & {error?: undefined} | {error: string}> {
 
     const packagePath = `${packageDir}/package.json`;
-    if(!fs.existsSync(packagePath)){
+    if(!await util.promisify(fs.exists)(packagePath)){
```

Comment:
not super related - thoughts on using `fsPromises` instead of `promisify`? the docs suggest `fsPromises.access` as an alternative for `exists`, and there is `fsPromises.readFile`

Reply:
is that a separate package? i just did what was easiest. we can easily change it.

Reply:
i also tried to use access via promisfy but it returns void and i guess throws if it doesn't exist. which is a pretty bad api.